### PR TITLE
[MXNET-299] Adding the Download page and linking it to the Install page

### PR DESCRIPTION
**Pull request by kpmurali**
## Description ##
Adding a download page with Apache mirror source links.

### Changes ###
- [ x  ] Adding a download page on the mxnet website. 
- [ x ] Download point to our mirror system, not gihtub managed links. ß--- ASF mirror

http://54.210.6.225/install/download.html #

### CODE REVIEW

**Comment by marcoabreu** [comment]
Hi @kpmurali, thanks for addressing the mentors comments. 

Just to clarify, these are source downloads, right? I think it would be a bit misleading to our users to classify this as just "Download" without any further information. At any point in time, I'd prefer our users to see the binary release as the first option besides the download of a source release - otherwise people might get a bit lost.

@aaronmarkham wdyt?

**Comment by kpmurali** [comment]
Hi @marcoabreu , yes these are links to the source downloads. @aaronmarkham, any suggestions on where else the link to the Download page can be placed?

File: docs/install/download.md [hunk 1]
```diff
@@ -1,8 +1,8 @@
-# Download
+# Source Download
 
-Choose your flavor of download from the following links:
+These source archives are generated from tagged releases. Updates and patches will not have been applied. For any updates refer to the corresponding branches in the GitHub repository. Choose your flavor of download from the following links:
```
Comment (marcoabreu) [inline docs/install/download.md]: Maybe make "GitHub repository" a link?

File: docs/install/index.md [hunk 2]
```diff
@@ -1743,4 +1744,10 @@ Will be available soon.
 </div>
 </div>
 
-# Download Source Package
+<br/>
+<!-- Download -->
+
+# Source Download
```
Comment (eric-haibin-lin) [inline docs/install/index.md]: @aaronmarkham the git clone section always clones the bleeding edge master. Should we move this "Source Download" to step 4 instead?

**Comment by anirudh2290** [comment]
@eric-haibin-lin can we merge this ?

**Pull request merged**